tools/lambda-promtail/lambda-promtail/main.go

```diff
@@ -68,6 +82,33 @@ func init() {
 	s3Clients = make(map[string]*s3.Client)
 }
 
+func parseExtraLabels(extraLabelsRaw string) ([]ExtraLabel, error) {
+	var extractedLabels []ExtraLabel
+	extraLabelsSplit := strings.Split(extraLabelsRaw, ",")
+
+	if len(extraLabelsRaw) < 1 {
+		return extractedLabels, nil
+	}
+
+	if len(extraLabelsSplit)%2 != 0 {
+		return nil, fmt.Errorf(invalidExtraLabelsError)
+	}
+	for i := 0; i < len(extraLabelsSplit); i += 2 {
+		extractedLabels = append(extractedLabels, ExtraLabel{
+			key:   extraLabelsSplit[i],
+			value: extraLabelsSplit[i+1],
+		})
+	}
+	fmt.Println("extra labels:", extractedLabels)
+	return extractedLabels, nil
+}
+
+func applyExtraLabels(labels model.LabelSet) {
```

Comment:
for some extra safety we could call `labels.IsValid()` at the end of this function, and have some basic tests for it as well?

Reply:
I updated the logic in `parseExtraLabels` to leverage the `model.LabelSet`. This allows me to simply rely on the existing `model.LabelSet.Validate` method to validate the extra label set and the `model.LabelSet.Merge` to add the extra labels.